File: flutter/lib/models/speech_segment.dart
import 'dart:typed_data';

/// A detected speech segment from VAD.
class SpeechSegment {
  final Float32List samples;
  final double startTime;
  final double endTime;

  SpeechSegment({
    required this.samples,
    required this.startTime,
    required this.endTime,
  });

  double get duration => endTime - startTime;
}

File: kmp/shared/src/commonMain/kotlin/com/sondt/justtranscribe/SpeechSegment.kt
package com.sondt.justtranscribe

/**
 * A finalized speech utterance emitted by [Segmenter]: float PCM samples
 * (mono, the configured sample rate, range -1..1) plus its position in the stream.
 */
class SpeechSegment(
    val samples: FloatArray,
    val startTime: Double,
    val endTime: Double,
) {
    val duration: Double get() = endTime - startTime
}
